src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/domainEvents/OutboundEventsPublisher.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.domainEvents
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.domainEvents.DomainEventsService.HMPPSDomainEvent
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+
+@Service
+class OutboundEventsPublisher(
+  private val hmppsQueueService: HmppsQueueService,
+  private val objectMapper: ObjectMapper,
+) {
+
+  companion object {
+    const val DOMAIN_TOPIC_ID = "domainevents"
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  private val domainEventsTopic by lazy {
+    hmppsQueueService.findByTopicId(DOMAIN_TOPIC_ID)
+      ?: throw NoSuchElementException("Topic with name $DOMAIN_TOPIC_ID doesn't exist")
+  }
+  private val domainEventsTopicClient by lazy { domainEventsTopic.snsClient }
+
+  fun publishDomainEvent(event: HMPPSDomainEvent, licenceId: String) {
+    log.debug("Event {} for id {}", event.eventType, licenceId)
+    val eventType = event.eventType
```

Comment:
super minor, would suggest either referring to `event.eventType` everywhere or pushing it up a line and referring to it in the debug log line and when building up the message attributes